### ISSUE TITLE
Properly respect XDG_CONFIG_HOME base directory specification

### ISSUE THREAD

**Issue by dsifford** [jonls/redshift#606]
Hi there,

I'm wondering if it would be possible to manipulate the config-ini.c file so that the `redshift.conf` file is read from the proper locations, per the XDG base directory specification.

https://github.com/jonls/redshift/blob/91e1d30a6f8f37e8a20575ac5e04dd0228080783/src/config-ini.c#L64-L68

If I'm reading this correctly, it appears that the config is read from the root of `XDG_CONFIG_HOME` folder, and not `XDG_CONFIG_HOME/redshift`, as it should per the specification. Is there any way this can be adjusted? This is especially important since the `redshift` directory already exists due to the somewhat recent addition of hook functionality.

**TL;DR:** `$XDG_CONFIG_HOME/redshift.conf` should instead be `$XDG_CONFIG_HOME/redshift/redshift.conf`.

Thanks for considering.

**Comment by jonls** [comment]
Thanks for the suggestion. I think, if this is implemented, we should prefer `$XDG_CONFIG_HOME/redshift/redshift.conf` but fall back to the current location at `$XDG_CONFIG_HOME/redshift.conf` if it's not found.

**Comment by dsifford** [comment]
100% agree. :+1:

**Comment by ILMostro** [comment]
Ahh, I was just about to point this out, as the current version on Gentoo still doesn't have this in place.  
Thanks for the great work.
